DashAI/front/src/components/experiments/PrepareDatasetStep.jsx

```diff
@@ -0,0 +1,398 @@
+    } else {
+      setSplitsReady(false);
+    }
+  }, [rowsPartitionsIndex, rowsPartitionsPercentage, rowsPartitionsError]);
+  useEffect(() => {
+    if (columnsReady && splitsReady) {
+      setNewExp({
+        ...newExp,
+        input_columns: inputColumns,
+        output_columns: outputColumns,
+        splits:
+          rowsPreference === "manually"
+            ? rowsPartitionsIndex
+            : rowsPartitionsPercentage,
+      }); // splits should depend on preference
+      setNextEnabled(true);
+    } else {
+      setNextEnabled(false);
+    }
+  }, [columnsReady, splitsReady]);
+
+  return (
+    <React.Fragment>
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Typography item variant="subtitle1" component="h3" sx={{ mb: 0 }}>
+            Choose which columns to use
+          </Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Typography
+            item
+            variant="caption"
+            component="h3"
+            sx={{ mb: 2, color: "grey" }}
+          >
+            Ex: 1-6, 23-108
+          </Typography>
+        </Grid>
+
+        <TextField
+          required
+          id="dataset-input-columns"
+          label="Input"
+          fullWidth
+          autoComplete="off"
+          onChange={handleInputColumnsChange}
+          error={parseInputColumnsError}
+          helperText={parseInputColumnsError ? parseInputColumnsErrorText : ""}
+          sx={{ mb: 2 }}
+        />
+        <TextField
+          required
+          id="dataset-output-columns"
+          label="Output"
+          fullWidth
+          autoComplete="off"
+          onChange={handleOutputColumnsChange}
+          error={parseOutputColumnsError}
+          helperText={
+            parseOutputColumnsError ? parseOutputColumnsErrorText : ""
+          }
+          sx={{ mb: 2 }}
+        />
+        <Grid container spacing={1}>
+          <Grid item xs={12}>
+            <Typography variant="subtitle1" component="h3" sx={{ mb: 2 }}>
+              Choose how you wanna divide the dataset for the experiment
+            </Typography>
+          </Grid>
+        </Grid>
+        <RadioGroup
+          aria-labelledby="demo-radio-buttons-group-label"
+          defaultValue={"random"}
+          name="radio-buttons-group"
+          onChange={handleRowsPreferenceChange}
+        >
+          <FormControlLabel
+            value="random"
+            control={<Radio />}
+            label="Use random rows by percentage"
+          />
+          {rowsPreference === "random" ? (
+            <React.Fragment>
+              <Grid container direction="row" spacing={4}>
+                <Grid item sx={{ xs: 4 }}>
+                  <TextField
+                    id="training"
+                    label="Training"
+                    autoComplete="off"
+                    error={rowsPartitionsError}
+                    onChange={handleRowsChange}
+                  />
```

Comment:
Some input comments: 

- You can use certain types to render a specific component for the selected type, for example number (type param).
- I think that perhaps compact inputs would look better (size param).
- You could add start and end adornments by using InputProps.

```suggestion
                  <TextField
                    id="training"
                    label="Training"
                    autoComplete="off"
                    type="number"
                    size="small"
                    InputProps={{
                      startAdornment: <InputAdornment position="end">%</InputAdornment>,
                    }}
                    error={rowsPartitionsError}
                    onChange={handleRowsChange}
                  />
```

If you change it, remember to change them all!